pandas/tests/test_categorical.py

```diff
@@ -3640,6 +3640,153 @@ def test_cat_accessor_no_new_attributes(self):
+            ('rindex', (" ",), {}),
+            ('rjust', (10,), {}),
+            ('rpartition', (" ",), {"expand": False}), # not default
+            ('rpartition', (" ",), {"expand": True}), # default
+            ('rsplit', (), {}),
+            ('rstrip', (), {}),
+            ('slice', (0,1), {}),
+            ('slice_replace', (0,1,"z"), {}),
+            ('split', (" ",), {"expand":False}), #default
+            ('split', (" ",), {"expand":True}), # not default
+            ('startswith', ("a",), {}),
+            ('strip', (), {}),
+            ('swapcase', (), {}),
+            ('title', (), {}),
+            #('translate', (), {}), #py2/3 specific arguments. WTF?
+            ('upper', (), {}),
+            ('wrap', (2,), {}),
+            ('zfill', (10,), {})
+        ]
+
+        # We can't make a categorical with lists as individual categories.
+        # -> `s.str.split(" ").astype("category")` will error!
+        # That's why the following two functions are never used for categories:
+        # * get
+        # * join
+
+        for func, args, kwargs in func_defs:
+            res = getattr(c.str, func)(*args, **kwargs)
+            exp = getattr(s.str, func)(*args, **kwargs)
+            if isinstance(res, pd.DataFrame):
+                tm.assert_frame_equal(res, exp)
+            else:
+                tm.assert_series_equal(res, exp)
+
+        invalid = Series([1,2,3]).astype('category')
+        with tm.assertRaisesRegexp(AttributeError, "Can only use .str accessor with string"):
+            invalid.str
+        self.assertFalse(hasattr(invalid, 'str'))
+
+    def test_dt_accessor_api_for_categorical(self):
+        # https://github.com/pydata/pandas/issues/10661
+        from pandas.tseries.common import Properties
+        from pandas.tseries.index import date_range
+        s = Series(date_range('1/1/2015', periods=5, tz="MET"))
+        c = s.astype("category")
+
+        self.assertIsInstance(c.dt, Properties)
+
+        func_defs = [
+            ('normalize', (), {}),
+            ('strftime', ("%Y-%m-%d",), {}),
+            ('to_period', (), {}),
+            ('to_pydatetime', (), {}), # available for DateTime, but not delta...
+            ('tz_convert', ("EST",), {}),
+            #('tz_localize', ("UTC",), {}), # the series is already localized
+        ]
+
+        for func, args, kwargs in func_defs:
+            try:
+                res = getattr(c.dt, func)(*args, **kwargs)
+                exp = getattr(s.dt, func)(*args, **kwargs)
+            except Exception as e:
+                print(func)
+                raise(e)
+
+            if isinstance(res, pd.DataFrame):
+                tm.assert_frame_equal(res, exp)
+            elif isinstance(res, pd.Series):
+                tm.assert_series_equal(res, exp)
+            else:
+                tm.assert_numpy_array_equal(res, exp)
+
+        attr_defs = ['date', 'day', 'dayofweek', 'dayofyear', 'days_in_month', 'daysinmonth',
```

Comment:
all of these are defined in ``DatetimeIndex._datetimelike_ops`` (and ``Timesdelta/PeriodIndex``), so no need to actually list them (and defeats the purpose if things are added). Just iterate the already existing list, providing special cases if needed.